pkg/storage/etcd3/etcdwatcher/event.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcdwatcher
```

Comment:
I think those are strictly related (it basically the same etcdv3 API).

Another point why I would like to move it to the etcd3 package is that, nothing from this package should be public. And with this separation you need to make some things public to make them available from the etcd3 package, which in my opinion is not the best choice.